## **Title -: ELB Sales Analysis**

#### **Questions -**
1. What was the highest-selling product in September?
2. Which product had the highest sales for the "CND Chennai" sales team in May?
3. Which customer had the maximum stock returns in June for the Bangalore HQ?
4. Which sales team had the maximum percentage of primary sales returned due to expiry?
5. What percentage of overall primary sales was a􀆯ected by breakage?
6. What were the primary sales for the Delhi HQ in the month of September?
7. What were the sales of "Britorva 20" in September for PALEPU PHARMA DIST PVT LTD under Coimbatore HQ?

#### **Bonus -**
1. What is the forecasted value of primary sales for the October month?
2. Which product had the highest forecasted sales for the "CND Chennai" sales team in November?
3. What were the forecasted sales of "Britorva 20" in October under Coimbatore HQ?

### Introduction

This report answers specific business questions using the ELB-Sales-Data dataset. The analysis focuses on identifying key metrics such as the highest-selling product, sales team performance, and the impact of returns.


### Import Libraries

In [6]:
# Importing the basic libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [7]:
# set display for all rows and columns 
pd.set_option("display.max_rows", None)
pd.set_option("display.max_columns", None)

### Import Dataset

In [9]:
# Importing the ELB excel file
df = pd.read_excel("ELB-Sales-Data.xlsx")

In [10]:
# read the first five rows of dataset
df.head()

Sr               ID                            Customer  Primary Sales  \
0  9354  SL-10-24-268441  SESHABALAJEE MEDISOLUTIONS PVT LTD         135.08   
1  9355  SL-10-24-268440  SESHABALAJEE MEDISOLUTIONS PVT LTD         413.80   
2  9356  SL-10-24-268439  SESHABALAJEE MEDISOLUTIONS PVT LTD         210.38   
3  9357  SL-10-24-268438  SESHABALAJEE MEDISOLUTIONS PVT LTD         105.19   
4  9358  SL-10-24-268437  SESHABALAJEE MEDISOLUTIONS PVT LTD         978.30   

   Cancelled  Is added to primary log  Against Expiry  Breakage  Claim Offer  \
0          1                        1          135.08       0.0          0.0   
1          1                        1          413.80       0.0          0.0   
2          1                        1          210.38       0.0          0.0   
3          1                        1          105.19       0.0          0.0   
4          1                        1          978.30       0.0          0.0   

   Quantity  Net Rate  Free Item        HQ Sales Team Return for Reason  \
0        -4     33.77          0  HQ-Vizag  Elbrit AP           Expired   
1        -4    103.45          0  HQ-Vizag  Elbrit AP           Expired   
2        -2    105.19          0  HQ-Vizag  Elbrit AP           Expired   
3        -1    105.19          0  HQ-Vizag  Elbrit AP           Expired   
4        -9    108.70          0  HQ-Vizag  Elbrit AP           Expired   

   Sales Return Sales Invoice Name  Rate Difference         Item Name  \
0           0.0      SINV-24-01509              0.0        TELBRIT 20   
1           0.0      SINV-24-01509              0.0         DAFAX M 5   
2           0.0      SINV-24-01509              0.0        TELBRIT AM   
3           0.0      SINV-24-01509              0.0         TELBRIT H   
4           0.0      SINV-24-01509              0.0  TRIGLIMIBRIT 1.3   

   Free Item Value       Date  
0              0.0 2024-09-23  
1              0.0 2024-09-23  
2              0.0 2024-09-23  
3              0.0 2024-09-23  
4              0.0 2024-09-23

### Data Cleaning and Preprocessing

In [12]:
df.head(2)

Sr               ID                            Customer  Primary Sales  \
0  9354  SL-10-24-268441  SESHABALAJEE MEDISOLUTIONS PVT LTD         135.08   
1  9355  SL-10-24-268440  SESHABALAJEE MEDISOLUTIONS PVT LTD         413.80   

   Cancelled  Is added to primary log  Against Expiry  Breakage  Claim Offer  \
0          1                        1          135.08       0.0          0.0   
1          1                        1          413.80       0.0          0.0   

   Quantity  Net Rate  Free Item        HQ Sales Team Return for Reason  \
0        -4     33.77          0  HQ-Vizag  Elbrit AP           Expired   
1        -4    103.45          0  HQ-Vizag  Elbrit AP           Expired   

   Sales Return Sales Invoice Name  Rate Difference   Item Name  \
0           0.0      SINV-24-01509              0.0  TELBRIT 20   
1           0.0      SINV-24-01509              0.0   DAFAX M 5   

   Free Item Value       Date  
0              0.0 2024-09-23  
1              0.0 2024-09-23

In [13]:
# shape of rows and columns
df.shape

(107444, 21)

In [14]:
# display the column names
df.columns

Index(['Sr', 'ID', 'Customer', 'Primary Sales', 'Cancelled',
       'Is added to primary log', 'Against Expiry', 'Breakage', 'Claim Offer',
       'Quantity', 'Net Rate', 'Free Item', 'HQ', 'Sales Team',
       'Return for Reason', 'Sales Return', 'Sales Invoice Name',
       'Rate Difference', 'Item Name', 'Free Item Value', 'Date'],
      dtype='object')

In [15]:
# details information of dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 107444 entries, 0 to 107443
Data columns (total 21 columns):
 #   Column                   Non-Null Count   Dtype         
---  ------                   --------------   -----         
 0   Sr                       107444 non-null  int64         
 1   ID                       107444 non-null  object        
 2   Customer                 107444 non-null  object        
 3   Primary Sales            107444 non-null  float64       
 4   Cancelled                107444 non-null  int64         
 5   Is added to primary log  107444 non-null  int64         
 6   Against Expiry           107444 non-null  float64       
 7   Breakage                 107444 non-null  float64       
 8   Claim Offer              107444 non-null  float64       
 9   Quantity                 107444 non-null  int64         
 10  Net Rate                 107444 non-null  float64       
 11  Free Item                107444 non-null  int64         
 12  HQ              

In [16]:
# checking null and missing values
df.isnull().sum()

Sr                             0
ID                             0
Customer                       0
Primary Sales                  0
Cancelled                      0
Is added to primary log        0
Against Expiry                 0
Breakage                       0
Claim Offer                    0
Quantity                       0
Net Rate                       0
Free Item                      0
HQ                           180
Sales Team                   180
Return for Reason          89300
Sales Return                   0
Sales Invoice Name             2
Rate Difference                0
Item Name                      0
Free Item Value                0
Date                           0
dtype: int64

In [17]:
df.isnull().mean()*100

Sr                          0.000000
ID                          0.000000
Customer                    0.000000
Primary Sales               0.000000
Cancelled                   0.000000
Is added to primary log     0.000000
Against Expiry              0.000000
Breakage                    0.000000
Claim Offer                 0.000000
Quantity                    0.000000
Net Rate                    0.000000
Free Item                   0.000000
HQ                          0.167529
Sales Team                  0.167529
Return for Reason          83.113064
Sales Return                0.000000
Sales Invoice Name          0.001861
Rate Difference             0.000000
Item Name                   0.000000
Free Item Value             0.000000
Date                        0.000000
dtype: float64

In [18]:
# drop 'Return for Reason' column 
df.drop(columns=['Return for Reason'], inplace=True)

In [19]:
df.head(2)

Sr               ID                            Customer  Primary Sales  \
0  9354  SL-10-24-268441  SESHABALAJEE MEDISOLUTIONS PVT LTD         135.08   
1  9355  SL-10-24-268440  SESHABALAJEE MEDISOLUTIONS PVT LTD         413.80   

   Cancelled  Is added to primary log  Against Expiry  Breakage  Claim Offer  \
0          1                        1          135.08       0.0          0.0   
1          1                        1          413.80       0.0          0.0   

   Quantity  Net Rate  Free Item        HQ Sales Team  Sales Return  \
0        -4     33.77          0  HQ-Vizag  Elbrit AP           0.0   
1        -4    103.45          0  HQ-Vizag  Elbrit AP           0.0   

  Sales Invoice Name  Rate Difference   Item Name  Free Item Value       Date  
0      SINV-24-01509              0.0  TELBRIT 20              0.0 2024-09-23  
1      SINV-24-01509              0.0   DAFAX M 5              0.0 2024-09-23

In [20]:
# fill the missing values of 'HQ', 'Sales Team' and 'Sales Invoice Name'
df['HQ'] = df['HQ'].fillna(df['HQ'].mode()[0])
df['Sales Team'] = df['Sales Team'].fillna(df['Sales Team'].mode()[0])
df['Sales Invoice Name'] = df['Sales Invoice Name'].fillna(df['Sales Invoice Name'].mode()[0])

In [21]:
df.isnull().sum()

Sr                         0
ID                         0
Customer                   0
Primary Sales              0
Cancelled                  0
Is added to primary log    0
Against Expiry             0
Breakage                   0
Claim Offer                0
Quantity                   0
Net Rate                   0
Free Item                  0
HQ                         0
Sales Team                 0
Sales Return               0
Sales Invoice Name         0
Rate Difference            0
Item Name                  0
Free Item Value            0
Date                       0
dtype: int64

In [22]:
df.head(2)

Sr               ID                            Customer  Primary Sales  \
0  9354  SL-10-24-268441  SESHABALAJEE MEDISOLUTIONS PVT LTD         135.08   
1  9355  SL-10-24-268440  SESHABALAJEE MEDISOLUTIONS PVT LTD         413.80   

   Cancelled  Is added to primary log  Against Expiry  Breakage  Claim Offer  \
0          1                        1          135.08       0.0          0.0   
1          1                        1          413.80       0.0          0.0   

   Quantity  Net Rate  Free Item        HQ Sales Team  Sales Return  \
0        -4     33.77          0  HQ-Vizag  Elbrit AP           0.0   
1        -4    103.45          0  HQ-Vizag  Elbrit AP           0.0   

  Sales Invoice Name  Rate Difference   Item Name  Free Item Value       Date  
0      SINV-24-01509              0.0  TELBRIT 20              0.0 2024-09-23  
1      SINV-24-01509              0.0   DAFAX M 5              0.0 2024-09-23

In [23]:
# checking duplicated rows from dataset
df.duplicated().sum()

0

### Exploratory Data Analysis and Data Visualization

In [25]:
df.head(2)

Sr               ID                            Customer  Primary Sales  \
0  9354  SL-10-24-268441  SESHABALAJEE MEDISOLUTIONS PVT LTD         135.08   
1  9355  SL-10-24-268440  SESHABALAJEE MEDISOLUTIONS PVT LTD         413.80   

   Cancelled  Is added to primary log  Against Expiry  Breakage  Claim Offer  \
0          1                        1          135.08       0.0          0.0   
1          1                        1          413.80       0.0          0.0   

   Quantity  Net Rate  Free Item        HQ Sales Team  Sales Return  \
0        -4     33.77          0  HQ-Vizag  Elbrit AP           0.0   
1        -4    103.45          0  HQ-Vizag  Elbrit AP           0.0   

  Sales Invoice Name  Rate Difference   Item Name  Free Item Value       Date  
0      SINV-24-01509              0.0  TELBRIT 20              0.0 2024-09-23  
1      SINV-24-01509              0.0   DAFAX M 5              0.0 2024-09-23

**1. What was the highest-selling product in September?**

Extract Data for September

In [28]:
# Ensure the 'Date' column is in datetime format
df['Date'] = pd.to_datetime(df['Date'])

# Filter data for September
september_data = df[df['Date'].dt.month == 9]

# Display September data to verify
september_data.head()

Sr               ID                            Customer  Primary Sales  \
0  9354  SL-10-24-268441  SESHABALAJEE MEDISOLUTIONS PVT LTD         135.08   
1  9355  SL-10-24-268440  SESHABALAJEE MEDISOLUTIONS PVT LTD         413.80   
2  9356  SL-10-24-268439  SESHABALAJEE MEDISOLUTIONS PVT LTD         210.38   
3  9357  SL-10-24-268438  SESHABALAJEE MEDISOLUTIONS PVT LTD         105.19   
4  9358  SL-10-24-268437  SESHABALAJEE MEDISOLUTIONS PVT LTD         978.30   

   Cancelled  Is added to primary log  Against Expiry  Breakage  Claim Offer  \
0          1                        1          135.08       0.0          0.0   
1          1                        1          413.80       0.0          0.0   
2          1                        1          210.38       0.0          0.0   
3          1                        1          105.19       0.0          0.0   
4          1                        1          978.30       0.0          0.0   

   Quantity  Net Rate  Free Item        HQ Sales Team  Sales Return  \
0        -4     33.77          0  HQ-Vizag  Elbrit AP           0.0   
1        -4    103.45          0  HQ-Vizag  Elbrit AP           0.0   
2        -2    105.19          0  HQ-Vizag  Elbrit AP           0.0   
3        -1    105.19          0  HQ-Vizag  Elbrit AP           0.0   
4        -9    108.70          0  HQ-Vizag  Elbrit AP           0.0   

  Sales Invoice Name  Rate Difference         Item Name  Free Item Value  \
0      SINV-24-01509              0.0        TELBRIT 20              0.0   
1      SINV-24-01509              0.0         DAFAX M 5              0.0   
2      SINV-24-01509              0.0        TELBRIT AM              0.0   
3      SINV-24-01509              0.0         TELBRIT H              0.0   
4      SINV-24-01509              0.0  TRIGLIMIBRIT 1.3              0.0   

        Date  
0 2024-09-23  
1 2024-09-23  
2 2024-09-23  
3 2024-09-23  
4 2024-09-23

 Calculate Total Sales for Each Product

In [30]:
# Group by product and sum the sales
september_sales = september_data.groupby('Item Name')['Primary Sales'].sum()

# Display the aggregated sales
september_sales

Item Name
ACEBRIT MR               18514.80
ACEBRIT P                63959.76
AMOXIBRIT 625           329224.74
ARNIBLOC 100            162966.70
ARNIBLOC 50             381422.54
AURAGEST SR 200         195696.00
AURAGEST SR 300          -2082.84
BENISTAR 4               93255.95
BENISTAR 8               48585.67
BISOBRIT 2.5            102491.71
BISOBRIT 5               45690.01
BISOBRIT AM 5             3974.98
BISOBRIT T 2.5           48394.10
BISOBRIT T 5             62712.28
BRITORVA 10             191616.21
BRITORVA 10F            170308.77
BRITORVA 20              94506.07
BRITORVA 40             191482.76
BRITORVA 80             207407.80
BRITORVA CV 10          579993.77
BRITORVA CV 20            7483.20
BRITVIT                 374876.01
BRITVOG 0.2              78902.64
BRITVOG 0.3             145376.50
BRITVOG M 0.2            30318.51
BRITVOG M 0.3            30874.38
C FERT                  414081.06
C FERT M                660362.28
CALBRIT                 451065.15
CALB

Identify the Highest-Selling Product

In [32]:
# Find the product with the highest sales
highest_selling_product = september_sales.idxmax()
highest_sales = september_sales.max()

print(f"The highest-selling product in September is '{highest_selling_product}' with sales of {highest_sales}.")

The highest-selling product in September is 'NEURONZ D' with sales of 2308078.58.


**2. Which product had the highest sales for the "CND Chennai" sales team in May?**

Filter Data for May and "CND Chennai"

In [35]:
# Filter for the month of May
may_data = df[df['Date'].dt.month == 5]

# Further filter for "CND Chennai" sales team
may_data_cnd = may_data[may_data['Sales Team'] == 'CND Chennai']

# Display filtered data to verify
may_data_cnd.head()

Sr               ID         Customer  Primary Sales  Cancelled  \
74731  102409  SL-05-24-133486  AADITYA PHARMEX            0.0          0   
74732  102410  SL-05-24-133485  AADITYA PHARMEX            0.0          0   
74733  102411  SL-05-24-133484  AADITYA PHARMEX         5946.5          0   
74734  102412  SL-05-24-133483  AADITYA PHARMEX            0.0          0   
74735  102413  SL-05-24-133482  AADITYA PHARMEX            0.0          0   

       Is added to primary log  Against Expiry  Breakage  Claim Offer  \
74731                        1             0.0       0.0          0.0   
74732                        1             0.0       0.0          0.0   
74733                        1             0.0       0.0          0.0   
74734                        1             0.0       0.0          0.0   
74735                        1             0.0       0.0          0.0   

       Quantity  Net Rate  Free Item          HQ   Sales Team  Sales Return  \
74731         5      0.00          1  HQ-Chennai  CND Chennai           0.0   
74732       140      0.00          1  HQ-Chennai  CND Chennai           0.0   
74733        50    118.93          0  HQ-Chennai  CND Chennai           0.0   
74734         5      0.00          1  HQ-Chennai  CND Chennai           0.0   
74735        60      0.00          1  HQ-Chennai  CND Chennai           0.0   

      Sales Invoice Name  Rate Difference     Item Name  Free Item Value  \
74731      IN-CH24-00692              0.0     NEBILOC 5      -594.642857   
74732      IN-CH24-00692              0.0     NEURONZ D    -16200.000000   
74733      IN-CH24-00692              0.0     NEBILOC 5         0.000000   
74734      IN-CH24-00692              0.0         MYGUT      -457.627119   
74735      IN-CH24-00692              0.0  NEURONZ PLUS     -8485.714286   

            Date  
74731 2024-05-31  
74732 2024-05-31  
74733 2024-05-31  
74734 2024-05-31  
74735 2024-05-31

Calculate Total Sales for Each Product

In [37]:
# Group by product and sum the sales
may_sales_cnd = may_data_cnd.groupby('Item Name')['Primary Sales'].sum()

# Display the aggregated sales
may_sales_cnd

Item Name
BRITVIT                 201766.82
BRITVOG M 0.2             4773.45
BRITVOG M 0.3            -2086.62
CALBRIT                  81119.30
CILNITAB 10             102845.73
CILNITAB 20              57732.29
CILNITAB 5               44348.66
CITIBRIT FORTE          103704.78
CITIBRIT PLUS            -9380.00
DABITON 110               9843.53
DAFAX V                  98499.84
DAFAX VM 500             34713.60
GLIMIBRIT M 0.5          34048.56
GLIMIBRIT M1 FORTE       15576.59
GLIMIBRIT M2 FORTE       29121.42
GLIZATO 30 OD            49978.60
GLIZATO 40               21937.00
GLIZATO 60 OD           172168.22
GLIZATO 80               85895.50
GLIZATO M40             143015.00
GLIZATO M80             404144.42
MYGUT                   105587.12
NEBILOC 10               41713.77
NEBILOC 2.5              30744.07
NEBILOC 5                62889.83
NEBILOC 5 AM             17396.80
NEURONZ D              1161546.19
NEURONZ PLUS            371374.20
ONLY E                   28697.94
ROZU

Identify the Product with the Highest Sales

In [39]:
# Find the product with the highest sales
highest_selling_product_may_cnd = may_sales_cnd.idxmax()
highest_sales_may_cnd = may_sales_cnd.max()

print(f"The product with the highest sales for 'CND Chennai' in May is '{highest_selling_product_may_cnd}' with sales of {highest_sales_may_cnd}.")

The product with the highest sales for 'CND Chennai' in May is 'NEURONZ D' with sales of 1161546.19.


**3. Which customer had the maximum stock returns in October for the Bangalore HQ?**

Filter Data for October and Bangalore HQ

In [42]:
# Filter for the month of October
october_data = df[df['Date'].dt.month == 6]

# Further filter for Bangalore HQ
october_bangalore_data = october_data[october_data['HQ'] == 'HQ-Bangalore']

# Display the filtered data to verify
october_bangalore_data.head()

Sr               ID         Customer  Primary Sales  Cancelled  \
60429  88107  SL-06-24-154736  MEDILINE BIOTEX         539.98          0   
60430  88108  SL-06-24-154735  MEDILINE BIOTEX        1035.02          0   
60431  88109  SL-06-24-154593  MEDILINE BIOTEX       11282.40          0   
60432  88110  SL-06-24-154592  MEDILINE BIOTEX        3297.90          0   
60433  88111  SL-06-24-154591  MEDILINE BIOTEX        3606.30          0   

       Is added to primary log  Against Expiry  Breakage  Claim Offer  \
60429                        1             0.0       0.0          0.0   
60430                        1             0.0       0.0          0.0   
60431                        1             0.0       0.0          0.0   
60432                        1             0.0       0.0          0.0   
60433                        1             0.0       0.0          0.0   

       Quantity  Net Rate  Free Item            HQ        Sales Team  \
60429        10     54.00          0  HQ-Bangalore  Elbrit Karnataka   
60430        10    103.50          0  HQ-Bangalore  Elbrit Karnataka   
60431        90    125.36          0  HQ-Bangalore  Elbrit Karnataka   
60432        30    109.93          0  HQ-Bangalore  Elbrit Karnataka   
60433        30    120.21          0  HQ-Bangalore  Elbrit Karnataka   

       Sales Return Sales Invoice Name  Rate Difference           Item Name  \
60429           0.0      IN-KA24-00717              0.0       CILNITAB T 40   
60430           0.0      IN-KA24-00717              0.0        ROZULA CV 20   
60431           0.0      IN-KA24-00715              0.0  SITADOC M 50 / 500   
60432           0.0      IN-KA24-00715              0.0      TRIGLIMIBRIT 1   
60433           0.0      IN-KA24-00715              0.0       OLMETOP 20 AM   

       Free Item Value       Date  
60429              0.0 2024-06-29  
60430              0.0 2024-06-29  
60431              0.0 2024-06-29  
60432              0.0 2024-06-29  
60433              0.0 2024-06-29

Calculate Total Returns for Each Customer

In [44]:
# Group by customer and sum the returns
october_returns_bangalore = october_bangalore_data.groupby('Customer')['Sales Return'].sum()

# Display the aggregated returns
october_returns_bangalore

Customer
DECCAN MEDISALES PVT LTD                   0.0
FOCUS MEDISALES PVT LTD BANASWADI          0.0
FOCUS MEDISALES PVT LTD HOSUR              0.0
FOCUS MEDISALES PVT LTD YESHWANTHPURA      0.0
HERITAGE MARKETEERS                        0.0
MAHAVEER MEDI SALES PVT LTD                0.0
MAHENDRA DRUG AGENCIES                     0.0
MEDILINE BIOTEX                            0.0
RSM PHARMA PVT LTD                         0.0
SAPTHAGIRI PHARMA DISTRIBUTORS             0.0
SHIVA DRUG HOUSE                           0.0
SRI PARSHVA PHARMA DISTRIBUTORS PVT LTD    0.0
SRINIVASA MEDISALES PVT LTD                0.0
VARDHAMAN MEDISALES PVT LTD YESWANTHPUR    0.0
VARDHMAN MEDISALES PVT LTD BANASWADI       0.0
VARDHMAN MEDISALES PVT LTD HO              0.0
Name: Sales Return, dtype: float64

In [45]:
october_returns_bangalore

Customer
DECCAN MEDISALES PVT LTD                   0.0
FOCUS MEDISALES PVT LTD BANASWADI          0.0
FOCUS MEDISALES PVT LTD HOSUR              0.0
FOCUS MEDISALES PVT LTD YESHWANTHPURA      0.0
HERITAGE MARKETEERS                        0.0
MAHAVEER MEDI SALES PVT LTD                0.0
MAHENDRA DRUG AGENCIES                     0.0
MEDILINE BIOTEX                            0.0
RSM PHARMA PVT LTD                         0.0
SAPTHAGIRI PHARMA DISTRIBUTORS             0.0
SHIVA DRUG HOUSE                           0.0
SRI PARSHVA PHARMA DISTRIBUTORS PVT LTD    0.0
SRINIVASA MEDISALES PVT LTD                0.0
VARDHAMAN MEDISALES PVT LTD YESWANTHPUR    0.0
VARDHMAN MEDISALES PVT LTD BANASWADI       0.0
VARDHMAN MEDISALES PVT LTD HO              0.0
Name: Sales Return, dtype: float64

Customer had the maximum stock returns in June for the Bangalore HQ is 0

**4. Which sales team had the maximum percentage of primary sales returned due to expiry?**

Calculate Percentage of Sales Returned Due to Expiry

In [49]:
# Calculate the percentage of primary sales returned due to expiry for each row
df['Expiry_Return_Percentage'] = (df['Against Expiry'] / df['Primary Sales']) * 100

# Handle cases where Primary Sales is zero to avoid division errors
df['Expiry_Return_Percentage'].fillna(0, inplace=True)
df['Expiry_Return_Percentage'] = df['Expiry_Return_Percentage'].replace([np.inf, -np.inf], 0)

# Display the updated DataFrame with the new column
print(df[['Sales Team', 'Primary Sales', 'Against Expiry', 'Expiry_Return_Percentage']].head())

  Sales Team  Primary Sales  Against Expiry  Expiry_Return_Percentage
0  Elbrit AP         135.08          135.08                     100.0
1  Elbrit AP         413.80          413.80                     100.0
2  Elbrit AP         210.38          210.38                     100.0
3  Elbrit AP         105.19          105.19                     100.0
4  Elbrit AP         978.30          978.30                     100.0


Group by Sales Team and Calculate the Average Percentage

In [51]:
# Group by sales team and calculate the average percentage
team_expiry_percentage = df.groupby('Sales Team')['Expiry_Return_Percentage'].mean()

# Display the calculated percentages for each team
team_expiry_percentage

Sales Team
Aura & Proxima Chennai        7.121484
Aura & Proxima Coimbatore    11.779880
Aura & Proxima Karnataka      5.988410
Aura & Proxima Kerala        14.156627
CND Chennai                  13.412698
CND Coimbatore               16.609769
Elbrit AP                     5.345729
Elbrit Chennai               12.939589
Elbrit Coimbatore            13.279207
Elbrit Delhi                  1.325411
Elbrit Karnataka             12.988705
Elbrit Kerala                20.311411
Elbrit Punjab                14.568880
Elbrit Punjab & Rajasthan     0.000000
Elbrit Rajasthan              7.702182
Elbrit Uttar Pradesh          4.009679
Vasco Chennai                 5.929487
Vasco Coimbatore              9.989181
Name: Expiry_Return_Percentage, dtype: float64

Identify the Sales Team with the Maximum Percentage

In [53]:
# Find the sales team with the maximum percentage
max_team = team_expiry_percentage.idxmax()
max_percentage = team_expiry_percentage.max()

print(f"The sales team with the maximum percentage of primary sales returned due to expiry is '{max_team}' with {max_percentage:.2f}%.")

The sales team with the maximum percentage of primary sales returned due to expiry is 'Elbrit Kerala' with 20.31%.


**5. What percentage of overall primary sales was affected by breakage?**

Calculate Total Breakage and Overall Primary Sales

In [56]:
# Calculate the total breakage and overall primary sales
total_breakage = df['Breakage'].sum()
total_primary_sales = df['Primary Sales'].sum()

# Display the totals
print(f"Total Breakage: {total_breakage}")
print(f"Total Primary Sales: {total_primary_sales}")

Total Breakage: -1051676.55
Total Primary Sales: 270157217.22


Calculate the Percentage of Primary Sales Affected by Breakage

In [58]:
# Calculate the percentage
breakage_percentage = -(total_breakage / total_primary_sales) * 100

# Display the result
print(f"The percentage of overall primary sales affected by breakage is {breakage_percentage:.2f}%.")

The percentage of overall primary sales affected by breakage is 0.39%.


**6. What were the primary sales for the Delhi HQ in the month of September?**

Filter Data for September and Delhi HQ

In [61]:
# Filter for the month of September
september_data = df[df['Date'].dt.month == 9]

# Further filter for Delhi HQ
september_delhi_data = september_data[september_data['HQ'] == 'HQ-Delhi']

# Display filtered data to verify
september_delhi_data.head()

Empty DataFrame
Columns: [Sr, ID, Customer, Primary Sales, Cancelled, Is added to primary log, Against Expiry, Breakage, Claim Offer, Quantity, Net Rate, Free Item, HQ, Sales Team, Sales Return, Sales Invoice Name, Rate Difference, Item Name, Free Item Value, Date, Expiry_Return_Percentage]
Index: []

Calculate Total Primary Sales

In [63]:
# Sum the primary sales for the filtered data
total_primary_sales_delhi_september = september_delhi_data['Primary Sales'].sum().astype(int)

print(f"Total primary sales for Delhi HQ in September is {total_primary_sales_delhi_september}")

Total primary sales for Delhi HQ in September is 0


**7. What were the sales of "Britorva 20" in September for PALEPU PHARMA DIST PVT LTD under Coimbatore HQ?**

Filter Data for September

In [66]:
# Filter for the month of September
september_data = df[df['Date'].dt.month == 9]

Further Filter for Coimbatore HQ

In [68]:
# Filter for Coimbatore HQ
coimbatore_data = september_data[september_data['HQ'] == 'HQ-Coimbatore']

Narrow Down to "PALEPU PHARMA DIST PVT LTD"

In [70]:
# Filter for the specific customer
palepu_pharma_data = coimbatore_data[coimbatore_data['Customer'] == 'PALEPU PHARMA DIST PVT LTD']

Check Sales of "Britorva 20"

In [72]:
# Filter for the product "Britorva 20"
britorva_data = palepu_pharma_data[palepu_pharma_data['Item Name'] == 'Britorva 20']

# Sum up the primary sales
britorva_sales = britorva_data['Primary Sales'].sum().astype(int)

print(f"Total sales of 'Britorva 20' in September for PALEPU PHARMA DIST PVT LTD under Coimbatore HQ is {britorva_sales}")

Total sales of 'Britorva 20' in September for PALEPU PHARMA DIST PVT LTD under Coimbatore HQ is 0


### **Bonus**

**1. What is the forecasted value of primary sales for the October month?**

In [75]:
from sklearn.linear_model import LinearRegression

In [76]:
# Extract year-month for aggregation
df['YearMonth'] = df['Date'].dt.to_period('M')

# Group by year-month and sum primary sales
monthly_sales = df.groupby('YearMonth')['Primary Sales'].sum()

# Convert to time series format
monthly_sales = monthly_sales.to_timestamp()

# Display the aggregated sales
print("Monthly Sales Data:")
print(monthly_sales)

Monthly Sales Data:
YearMonth
2024-04-01    43944346.66
2024-05-01    45896730.82
2024-06-01    47566563.99
2024-07-01    50067106.19
2024-08-01    41066121.82
2024-09-01    41616347.74
Freq: MS, Name: Primary Sales, dtype: float64


In [77]:
# Prepare data for regression
if len(monthly_sales) > 1:  # Ensure there is enough data for regression
    X = np.arange(len(monthly_sales)).reshape(-1, 1)  # Month indices as features
    y = monthly_sales.values  # Sales values as target

    # Train the linear regression model
    model = LinearRegression()
    model.fit(X, y)

    # Forecast sales for the next month (October)
    next_month = np.array([[len(monthly_sales)]])  # Index for October
    forecast = model.predict(next_month)

    print(f"Forecasted primary sales for October: {forecast[0]:.2f}")
else:
    print("Not enough data for Linear Regression forecasting.")

Forecasted primary sales for October: 42663074.93


**2. Which product had the highest forecasted sales for the "CND Chennai" sales team in November?**

In [79]:
from sklearn.linear_model import LinearRegression

In [80]:
# Filter for the "CND Chennai" sales team
cnd_chennai_data = df[df['Sales Team'] == 'CND Chennai']

In [81]:
# Extract year-month for aggregation
cnd_chennai_data['YearMonth'] = cnd_chennai_data['Date'].dt.to_period('M')

# Group by product and year-month, summing primary sales
product_monthly_sales = cnd_chennai_data.groupby(['Item Name', 'YearMonth'])['Primary Sales'].sum()

# Unstack to create a time series for each product
product_sales_ts = product_monthly_sales.unstack(level=0)

# Display the time series data
print(product_sales_ts)

Item Name    BRITVIT  BRITVOG M 0.2  BRITVOG M 0.3    CALBRIT  CILNITAB 10  \
YearMonth                                                                    
2024-04    310995.83       17268.06       36260.58  100870.14    105358.44   
2024-05    201766.82        4773.45       -2086.62   81119.30    102845.73   
2024-06    281207.26       55805.97       85287.15  132621.55    101424.37   
2024-07    236602.98       36972.54       29547.57  124283.98    135218.82   
2024-08    189072.89       17531.58        2330.25   71583.34     84872.73   
2024-09    178062.37       19925.41       11896.92   84081.10    162289.41   

Item Name  CILNITAB 20  CILNITAB 5  CITIBRIT FORTE  CITIBRIT PLUS  \
YearMonth                                                           
2024-04       34472.75    31415.88        81340.90            NaN   
2024-05       57732.29    44348.66       103704.78       -9380.00   
2024-06       45185.82    41059.56       130820.25      -10239.12   
2024-07       17756.12    5645

In [82]:
# Initialize a dictionary to store forecasts
forecasts = {}

# Iterate over each product's time series
for product in product_sales_ts.columns:
    # Select the time series for the product
    sales_ts = product_sales_ts[product].dropna()

    # Check if there is enough data for regression
    if len(sales_ts) > 1:  # Ensure at least two data points
        # Prepare data for regression
        X = np.arange(len(sales_ts)).reshape(-1, 1)
        y = sales_ts.values

        # Train a linear regression model
        model = LinearRegression()
        model.fit(X, y)

        # Predict sales for the next period (November)
        next_period = np.array([[len(sales_ts)]])
        forecast = model.predict(next_period)
        forecasts[product] = forecast[0]

# Display the forecasts
print(f"Forecasted sales for November: {forecasts}")

Forecasted sales for November: {'BRITVIT': 158216.02133333328, 'BRITVOG M 0.2': 28652.272666666664, 'BRITVOG M 0.3': 10775.248, 'CALBRIT': 87004.16999999998, 'CILNITAB 10': 141787.94666666668, 'CILNITAB 20': 4108.6600000000035, 'CILNITAB 5': 39218.49066666667, 'CITIBRIT FORTE': 100478.13200000001, 'CITIBRIT PLUS': 337.41333333333205, 'DABITON 110': 6320.735999999999, 'DAFAX V': 62806.546, 'DAFAX VM 500': -9507.669333333339, 'GLIMIBRIT M 0.5': 28522.471333333335, 'GLIMIBRIT M1 FORTE': 7462.751333333335, 'GLIMIBRIT M2 FORTE': 12018.998000000001, 'GLIZATO 30 OD': 33476.80133333334, 'GLIZATO 40': 22967.76066666667, 'GLIZATO 60 OD': 105462.32133333333, 'GLIZATO 80': 86523.366, 'GLIZATO M40': 93856.33600000001, 'GLIZATO M80': 351726.4046666667, 'MYGUT': 144373.75266666667, 'NEBILOC 10': 70000.79666666666, 'NEBILOC 2.5': 18569.624, 'NEBILOC 5': 50377.91, 'NEBILOC 5 AM': 20843.22466666667, 'NEURONZ D': 856423.9806666666, 'NEURONZ PLUS': 285119.6279999999, 'ONLY E': 36910.282666666666, 'ROZULA 

In [83]:
# Find the product with the maximum forecasted sales
highest_forecasted_product = max(forecasts, key=forecasts.get)
highest_forecasted_sales = forecasts[highest_forecasted_product]

print(f"The product with the highest forecasted sales for 'CND Chennai' in November is '{highest_forecasted_product}' with forecasted sales of {highest_forecasted_sales:.2f}")

The product with the highest forecasted sales for 'CND Chennai' in November is 'NEURONZ D' with forecasted sales of 856423.98


**3. What were the forecasted sales of "Britorva 20" in October under Coimbatore HQ?**

In [85]:
from sklearn.linear_model import LinearRegression

In [86]:
# Filter for Coimbatore HQ and "Britorva 20"
britorva_data = df[(df['HQ'] == 'HQ-Coimbatore') & (df['Item Name'] == 'Britorva 20')]

# Check if there is data available
if britorva_data.empty:
    print("No data available for 'Britorva 20' under Coimbatore HQ.")
else:
    print("Data for 'Britorva 20' under Coimbatore HQ is available.")

No data available for 'Britorva 20' under Coimbatore HQ.


In [87]:
# Extract year-month for aggregation
britorva_data['YearMonth'] = britorva_data['Date'].dt.to_period('M')

# Group by year-month and sum primary sales
monthly_sales = britorva_data.groupby('YearMonth')['Primary Sales'].sum()

# Convert to time series format
monthly_sales = monthly_sales.to_timestamp()

# Display the aggregated sales
print("Monthly Sales Data:")
print(monthly_sales)

Monthly Sales Data:
Series([], Name: Primary Sales, dtype: float64)


In [88]:
# Prepare data for regression
if len(monthly_sales) > 1:  # Ensure there is enough data for regression
    X = np.arange(len(monthly_sales)).reshape(-1, 1)  # Month indices as features
    y = monthly_sales.values  # Sales values as target

    # Train the linear regression model
    model = LinearRegression()
    model.fit(X, y)

    # Forecast sales for the next month (October)
    next_month = np.array([[len(monthly_sales)]])  # Index for October
    forecast = model.predict(next_month)

    print(f"Forecasted sales of 'Britorva 20' in October under Coimbatore HQ: {forecast[0]:.2f}")
else:
    print("Not enough data for Linear Regression forecasting.")

Not enough data for Linear Regression forecasting.


### Conclusions

#### Answers of 7 Questions :-

1. What was the highest-selling product in September?

=> The highest-selling product in September is 'NEURONZ D' with sales of 2308078.58

2. Which product had the highest sales for the "CND Chennai" sales team in May?

=> The product with the highest sales for 'CND Chennai' in May is 'NEURONZ D' with sales of 1161546.19

3. Which customer had the maximum stock returns in June for the Bangalore HQ?

=> Customer had the maximum stock returns in June for the Bangalore HQ is 0

4. Which sales team had the maximum percentage of primary sales returned due to expiry?

=> The sales team with the maximum percentage of primary sales returned due to expiry is 'Elbrit Kerala' with 20.31%.

5. What percentage of overall primary sales was affected by breakage?

=> The percentage of overall primary sales affected by breakage is 0.39%.

6. What were the primary sales for the Delhi HQ in the month of September?

=> Total primary sales for Delhi HQ in September is 0

7. What were the sales of "Britorva 20" in September for PALEPU PHARMA DIST PVT LTD under Coimbatore HQ?

=> Total sales of 'Britorva 20' in September for PALEPU PHARMA DIST PVT LTD under Coimbatore HQ is 0

#### Answers of Bonus Questions:-

1. What is the forecasted value of primary sales for the October month?

=> Forecasted primary sales for October is 42663074.93

2. Which product had the highest forecasted sales for the "CND Chennai" sales team in November?

=> The product with the highest forecasted sales for 'CND Chennai' in November is 'NEURONZ D' with forecasted sales of 856423.98

3. What were the forecasted sales of "Britorva 20" in October under Coimbatore HQ?

=> Incomplete Data, Not enough data for Linear Regression forecasting.